.github/workflows/ci.yml

```
@@ -739,12 +739,67 @@ jobs:
           name: ${{ env.artifacts_name }}
           path: artifacts
 
+  checkgovernance:
+    name: Check governance
+    runs-on: ubuntu-latest
+    needs: 'genmatrix'
+    outputs:
+      can_tag: ${{ steps.check-governance.outputs.can_tag }}
+    env:
+      GITHUB_REPOSITORY_OWNER: ${{ github.repository_owner }}
+      GITHUB_REPOSITORY: ${{ github.repository }}
+      GITHUB_REF_NAME: ${{ github.ref_name }}
+      MATRICES: ${{ needs.genmatrix.outputs.matrix }}
+    steps:
+
+      - name: Check governance
+        id: check-governance
+        run: |
+          CAN_TAG=0
+
+          # If we own it, then we can tag it
+          if [[ $GITHUB_REPOSITORY_OWNER == "silverstripe" ]]; then
+            echo "Repository is in the silverstripe organisation"
+            CAN_TAG=1
+          fi
+
+          # If we don't own it, we can only tag it if it's a supported module
+          if [[ $CAN_TAG == "0" ]]; then
+            # The installer version is the same for all generated matrices in a given run
+            CMS_VERSION=$(echo $MATRICES | jq -r '.include[0].installer_version')
+            # If the installer version wasn't an expected semver constraint, we're doing something weird
+            if [[ $CMS_VERSION =~ ^([0-9]+)(\.[0-9]+)?\.(x-dev$|[0-9]+(-(alpha|rc|beta)[0-9]+)?$) ]]; then
+              CMS_MAJOR=${BASH_REMATCH[1]}
+              # We don't really care if this is a "valid" branch per se (that is checked in gaugerelease) - we just need enough information
+              # to get what will be the major release portion of the branch name.
+              if [[ $GITHUB_REF_NAME =~ ^[0-9]+ ]]; then
+                BRANCH_MAJOR=${BASH_REMATCH[0]}
+                echo "Checking for supported modules in CMS $CMS_MAJOR against branch major $BRANCH_MAJOR"
+                curl -s -o __modules.json https://raw.githubusercontent.com/silverstripe/supported-modules/$CMS_MAJOR/modules.json
+                SUPPORTED_BRANCHES=$(jq -r --arg repo ${GITHUB_REPOSITORY} 'map(select(.github == $repo))[].branches[]' __modules.json)
+                SUPPORTED_BRANCH=$(echo $SUPPORTED_BRANCHES | grep -Po "$BRANCH_MAJOR" | head -n 1)
+                if [[ -n $SUPPORTED_BRANCH ]]; then
```

```suggestion
                if [[ $SUPPORTED_BRANCH != "" ]]; then
```
Standard for how we do this elsewhere for readability, double brackets makes this work, -n / -z makes more sense if using single brackets

---

Done